shared/chat/create-channel/container.tsx

```diff
@@ -63,7 +63,15 @@ export default Container.connect(
       channelname: string
       description: string
       teamID: TeamsTypes.TeamID
-    }) => dispatch(TeamsGen.createCreateChannel({channelname, description, teamID})),
+    }) =>
+      dispatch(
+        TeamsGen.createCreateChannel({
+          channelname,
+          description,
+          navToChatOnSuccess: Container.getRouteProps(ownProps, 'navToChatOnSuccess', true)!,
```

Comment:
Note ! Is just a typing annotation and not runtime anything

Reply:
Yeah that's why I recommended not using it.

Reply:
I disagree here - `getRouteProps` could be typed better (though I tried a little bit and it was hard enough that I gave up) - it can only return undefined when the third argument is undefiined, which it isn't here. I could instead do `getRouteProps(ownProps, 'nav...', undefined) ?? true` if you like that better. But this is guaranteed not to be undefined, which is why I used `!`.

Reply:
please don't change this helper as its going away very soon

Reply:
Where's it going?

Reply:
oh does nav 5 make this less cumbersome? great!